### ISSUE TITLE
data.table - change default: check.names=TRUE

### ISSUE THREAD

**Issue by jangorecki** [Rdatatable/data.table#4362]
in data.frame it is TRUE by default, not sure if it always was, but long enough.
R 3.1.0
```r
args(data.frame)
#function (..., row.names = NULL, check.rows = FALSE, check.names = TRUE, 
#    stringsAsFactors = default.stringsAsFactors()) 
```

in data.table
```r
args(data.table)
#function (..., keep.rownames = FALSE, check.names = FALSE, key = NULL, 
#    stringsAsFactors = FALSE)
```

**Comment by HughParsonage** [comment]
The convenience is being able to work with tables with strange names or work with tables without interacting directly with the names.  There might be occasions where you want a column to have a space or odd characters in it (for displaying, printing, charts) and functions that prepare such output might have `data.table()` within them and treat the column names agnostically.

**Comment by HughParsonage** [comment]
I don't think we should introduce a breaking change that is directly contrary to an advertised feature. The linked issues are only adventitious whereas `check.names = FALSE` is a feature.

**Comment by MichaelChirico** [comment]
> The convenience is being able to work with tables with strange names or work with tables without interacting directly with the names. There might be occasions where you want a column to have a space or odd characters in it (for displaying, printing, charts) and functions that prepare such output might have `data.table()` within them and treat the column names agnostically.

there's also a "less surprises" aspect of reading code like:

```r
data.table(`a b c d` = 1:3)
```

The resulting name is exactly what you see: `a b c d`. It's slightly less convenient to work with, but we don't need to memorize any rules about how `make.names` works to guess how the code will proceed. Consider also:

```r
data.frame(`a b c` = 1, a.b.c = 2, z=3, z=4)
#   a.b.c.1 a.b.c z z.1
# 1       1     2 3   4
```

Good luck guessing the `.1` suffix would go on columns 1 and 4!
